python/ray/data/_internal/execution/streaming_executor.py

```diff
@@ -110,6 +110,11 @@ def execute(
 
             logger.debug("Execution config: %s", self._options)
 
+            # Note: DAG must be initialized in order to query num_outputs_total.
```

Comment:
could you add a comment indicating why this must happen prior to operator bars being initialized, so readers know about this fix?